### PR TITLE
Add sort_complex

### PR DESCRIPTION
**Pull request by iliailmer**
Hi, 

This is an attempt to implement a `sort_complex` routine from `good_first_issue` category.

### CODE REVIEW

**Review by hawkinsp — COMMENTED** [review]
It looks like we now have two PRs for `sort_complex`, the other one being #3722 from a few days ago. We can only accept at most one of them :-(

I suggest that you combine forces: perhaps since the other PR was first, can you contribute some review suggestions on that one?

I took a quick look at this PR. A few general suggestions:
* it makes many formatting changes that aren't relevant to the PR itself (e.g., it reformats `lax_numpy.py`). In general, try to keep PRs focused on changing only one thing. If we did want to reformat `lax_numpy.py`, we would do that in a separate PR that only does reformatting without functional changes. So I'd revert the reformatting and make the new code match the existing code. Similarly for the change to `.gitignore`; I don't know whether it is a good change or not, but it is separate to what the PR does.

* your version of the PR does handle a few cases the other PR does not (e.g., `axis=None`). I'd suggest that the best path forward is to work with the other PR author to make the other PR incorporate your improvements. Alternatively we can merge that one and then you can rebase your PR on top of that one.

* nit: `.copy()` mostly isn't necessary in JAX, since JAX arrays are immutable.

Thanks!

File: jax/numpy/lax_numpy.py [hunk 1]
```diff
@@ -2555,1276 +2765,1415 @@ def wrapper(*args, **kwargs):
...
+        return argsort(a.ravel(), 0)
+    else:
+        axis = _canonicalize_axis(axis, ndim(a))
+        iota = lax.broadcasted_iota(np.int64, shape(a), axis)
+        _, perm = lax.sort_key_val(a, iota, dimension=axis)
+        return perm
+
+
+@_wraps(np.sort_complex)
+def sort_complex(a, axis=-1, kind="quicksort", order=None):
```
Comment (hawkinsp) [inline jax/numpy/lax_numpy.py:3398]: Actually, on reflection, as far as I can tell NumPy's API doesn't have any of the additional arguments here, only the array:
https://numpy.org/doc/stable/reference/generated/numpy.sort_complex.html

We should not deviate from NumPy without a good reason.

**Comment by gnecula** [comment]
@SIben FYI

File: jax/numpy/lax_numpy.py (hunk 1 above)
Comment (iliailmer) [reply jax/numpy/lax_numpy.py:3398]: Thank you for your comments, I appreciate it. I will fix the formatting, and I will contact the author of the other PR, hopefully we can join efforts on this. Thanks again!

**Comment by jakevdp** [comment]
`sort_complex` was merged in #3870

**Pull request closed without merging**